rpcs3/rpcs3qt/rsx_debugger.cpp

```diff
@@ -543,80 +641,544 @@ void rsx_debugger::GetMemory() const
 		m_list_captured_draw_calls->setItem(i, 0, new QTableWidgetItem(qstr(frame_debug.draw_calls[i].name)));
 }
 
+#pragma optimize( "", off )
 void rsx_debugger::GetBuffers() const
 {
 	const auto render = rsx::get_current_renderer();
-	if (!render)
+	if (!render || !render->is_initialized || !render->local_mem_size || !render->is_paused())
 	{
 		return;
 	}
 
-	// Draw Buffers
-	// TODO: Currently it only supports color buffers
-	for (u32 bufferId=0; bufferId < render->display_buffers_count; bufferId++)
+	auto addrs = render->get_color_surface_addresses();
+
+	// Color buffers
+	for (u32 buffer_it = 0; buffer_it < addrs.size(); buffer_it++)
 	{
-		const auto buffers = render->display_buffers;
-		const u32 rsx_buffer_addr = rsx::constants::local_mem_base + buffers[bufferId].offset;
+		const u32 rsx_buffer_addr = addrs[buffer_it];
+		const u32 width  = rsx::method_registers.surface_clip_width();
+		const u32 pitch = rsx::method_registers.surface_pitch(buffer_it);
+		const u32 height = rsx::method_registers.surface_clip_height();
+
+		Buffer* pnl{};
```

Comment:
What's a pnl?
If the word is already so short, abbreviations are super unnecessary.